web/src/model/filters.ts

```diff
@@ -160,7 +160,11 @@ export const removeFromFilters = (activeFilters: Filter[], search: FilterKey): F
 };
 
 export const filterKeyEqual = (f1: FilterKey, f2: FilterKey): boolean => {
-  return f1.def.id === f2.def.id && f1.not == f2.not && f1.moreThan == f2.moreThan;
+  return (
+    f1.def.id === f2.def.id &&
+    (f1.not === true) === (f2.not === true) &&
+    (f1.moreThan === true) === (f2.moreThan === true)
+  );
```

Comment:
why not
```
 return (
￼    f1.def.id === f2.def.id &&
￼    f1.not === f2.not &&
￼    f1.moreThan === f2.moreThan === true
￼  );
```

also should we check f1 and f2 at the begining 
```
if (!f1 || !f2) {
 /// generate an error 
}
```

Reply:
I convert each `not` / `moreThan` field to boolean first since [these are optionnal](https://github.com/netobserv/network-observability-console-plugin/blob/main/web/src/model/filters.ts#L70-L71) which means it can be either `undefined`, `false` or `true`.

If field are `undefined` or `false`, it should be considered as equal so `==` or `===` doesn't work.

`f1.moreThan === f2.moreThan === true` would only work if true
`if (!f1 || !f2)` is not necessary as `FilterKey` arguments are not optionnals

Reply:
An alternative would be to use `Boolean(a).valueOf()`